adapters/repos/db/lsmkv/bucket.go

```diff
@@ -495,58 +486,48 @@ func (b *Bucket) Shutdown(ctx context.Context) error {
 	}
 }
 
-func (b *Bucket) initFlushCycle(interval time.Duration) {
-	go func() {
-		t := time.Tick(interval)
-		for {
-			select {
-			case <-b.flushCycle.Stopped:
-				return
-			case <-t:
-				b.flushLock.Lock()
-
-				// to check the current size of the WAL to
-				// see if the threshold has been reached
-				stat, err := b.active.commitlog.file.Stat()
-				if err != nil {
-					b.logger.WithField("action", "lsm_wal_stat").
-						WithField("path", b.dir).
-						WithError(err).
-						Fatal("flush and switch failed")
-				}
-
-				memtableTooLarge := b.active.Size() >= b.memTableThreshold
-				walTooLarge := uint64(stat.Size()) >= b.walThreshold
-				dirtyButIdle := (b.active.Size() > 0 || stat.Size() > 0) &&
-					b.active.IdleDuration() >= b.flushAfterIdle
-				shouldSwitch := memtableTooLarge || walTooLarge || dirtyButIdle
-
-				// If true, the parent shard has indicated that it has
-				// entered an immutable state. During this time, the
-				// bucket should refrain from flushing until its shard
-				// indicates otherwise
-				if shouldSwitch && b.isReadOnly() {
-					b.logger.WithField("action", "lsm_memtable_flush").
-						WithField("path", b.dir).
-						Warn("flush halted due to shard READONLY status")
-
-					b.flushLock.Unlock()
-					time.Sleep(time.Second)
-					continue
-				}
-
-				b.flushLock.Unlock()
-				if shouldSwitch {
-					if err := b.FlushAndSwitch(); err != nil {
-						b.logger.WithField("action", "lsm_memtable_flush").
-							WithField("path", b.dir).
-							WithError(err).
-							Errorf("flush and switch failed")
-					}
-				}
-			}
+func (b *Bucket) initFlushCycle() {
```

Comment:
Does this name still make sense? If I see it correctly, the method no longer initiates a cycle, but rather makes a decision to flush if specific criteria are met. So maybe something along the lines of `maybeFlushAndSwitch` or `flushAndSwitchIfThresholdsMet`?